services/slam/builtin/builtin_test.go

```diff
@@ -1062,6 +1062,20 @@ func TestEndpointFailures(t *testing.T) {
 	test.That(t, fmt.Sprint(err), test.ShouldContainSubstring, "error getting the internal state from the SLAM client")
 	test.That(t, internalState, test.ShouldBeNil)
 
+	callbackPointCloud, err := svc.GetPointCloudMapStream(context.Background(), "hi")
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, callbackPointCloud, test.ShouldNotBeNil)
+	chunkPCD, err := callbackPointCloud()
+	test.That(t, err.Error(), test.ShouldContainSubstring, "Unimplemented desc")
```

Comment:
Why is the error here "Unimplemented desc"?

Reply:
The error generated is `Unimplemented desc` as these unit tests do not have a valid server for these streaming calls to ping As stated in one of the above comments, the creation of the client never fails unless context has been canceled (tested in `server_test.go`), therefore we need to ask for data to generate the error.

Reply:
"unimplemented desc" is the error that occurs when the callback function gets made but the actual endpoint does not exist, which is why the above error check on `1066` checks for nil. The endpoint does not fail until you request a chunk/until callbackPointCloud happens. 

originally this error was not wrapped because its the same error that triggers when EOF occurs, and we were unsure if  `errors.Is(err, io.EOF)` would trigger if the error was wrapped. After some testing it looks like that works, so I'll add a wrap for this error to test for